docs/boilerplate.md

```diff
@@ -0,0 +1,126 @@
+# Understanding Boilerplate in Oryx
+
+"How to Create Maintainable and Upgradeable Applications"
+
+## Introduction
+
+Boilerplate refers to the _template_ code that is used to generate an application code that can be further customized. Changing boilerplate is convenient at first sight as the code is at hand, generated in your project repository. However, when upgrading to newer versions of the original code, it becomes challenging. If you've customize the boilerplate code and want to reapply a new version of the boilerplate, you have to merge the customizations with the boilerplate during every update to a new version of the application. This is a time-consuming and error-prone process that can slow down your development process and increase the risk of bugs.
+
+## How we prevent boilerplate code
+
+In Oryx, we use a few tactics to prevent boilerplate code. These tactics include:
+
+1. bootstrap the application from npm packages rather than source.
+2. Expose an appBuilder function to setup the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a public repository ([github.com/spryker/oryx](https://github.com/spryker/oryx)) to enable developers to customize the code. The code is implemented with TypeScript and the types and interfaces are documented where ever needed to provide a good developer experience.
+5. Allow for configuration of business logic and component behavior.
+6. Allow for customizations by [dependency injection](./dependency-injection.md).
+
+Most of the details around these tactics can be found elsewhere in the documentation. In the following sections you read more details on the boilerplate files that are needed.
+
+### NPM Packages
+
+One of the biggest sources of boilerplate code in any project is the application logic itself. In Oryx, we've separated out all the application logic into individual packages that are distributed on npm. These packages include components, business logic and integrations to the Spryker APIs.
+
+By separating out the application logic in this way, we've eliminated the boilerplate code from your projects. Instead of writing code to handle basic functionality, you can simply install the appropriate package and use it directly in your code. And because the packages are distributed on npm and published with Semantic Versioning, you can be confident that upgrading to a new version of the framework won't break your code.
```

Comment:
```suggestion
By separating out the application logic in this way, we've eliminated the boilerplate code from your projects. Instead of writing code to handle basic functionality, you can simply install the appropriate package and use it directly in your code. And because the packages are distributed on npm and published with [Semantic Versioning](https://semver.org/), you can be confident that upgrading to a new version of the framework won't break your code.
```